File: sponge-standalone-app/examples/standalone/error_reporting_trigger/error_reporting_trigger.rb
# Sponge Knowledge Base
# Error reporting

class HelloWorldTrigger < Trigger
    def onConfigure
        self.withEvent("helloEvent")
    end

    def onRun(event)
        whatIsThis.doSomething()
        puts event.get("say")
    end
end

def onStartup
	$sponge.event("helloEvent").set("say", "Hello World!").send()
end

File: sponge-standalone-app/examples/standalone/error_reporting_trigger/error_reporting_trigger.js
/**
 * Sponge Knowledge Base 
 * Error reporting
 */

var HelloWorldTrigger = Java.extend(Trigger, {
    onConfigure: function(self) {
        self.withEvent("helloEvent");
    },
    onRun : function(self, event) {
        whatIsThis.doSomething();
        print(event.get("say"));
    }
});

function onStartup() {
    sponge.event("helloEvent").set("say", "Hello World!").send();
}
